spinnirker/subgraph.py

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2025 The University of Manchester
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Set
+from dataclasses import dataclass, field
+
+
+@dataclass(eq=False)
+class SubGraph():
+    """
+    This is a sub-graph of the NIR graph, which contains the names of the
+    nodes in the sub-graph.
+    """
+    nodes: Set[str] = field(default_factory=set)
```

Comment:
Not sure how this is used so these comments may not be valid.

1. Should it be an ordered set for run constancy
2. Should it be a public field